src/status.rs

```diff
@@ -517,10 +297,13 @@ impl StatusBuilder {
 
     /// Add an extension to the `Status` object.
     ///
-    /// The prefix `ext_` will automatically be prepended to the name during
-    /// serialization, if not already present.
+    /// The prefix `ext_` will automatically be prepended to the name, if not already present.
     pub fn add_extension<V: Into<Value>>(mut self, name: &str, value: V) -> Self {
-        self.extensions.insert(name.trim_start_matches("ext_").to_owned(), value.into());
+        if name.starts_with("ext_") {
+            self.extensions.insert(name.to_owned(), value.into());
+        } else {
+            self.extensions.insert(format!("ext_{}", name), value.into());
+        }
```

Comment:
This changes the in memory representation of the `self.extensions` field to always have the `ext_` prefix. Previously we added it when serializing. But we didn't remove it when de-serializing, so it was inconsistent anyway.

Reply:
Is this a breaking change? The field is public so I guess yes?

Reply:
I guess so, yes.

Reply:
Added to commit message.